File: LeetCode/String/Easy/344. Reverse String/main.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    void reverseString(vector<char> &s)
    {
        int l = 0;
        int r = s.size() - 1;

        while (l < r)
        {
            int temp = s[l];
            s[l] = s[r];
            s[r] = temp;
            r--;
            l++;
        }

        for (auto i = s.begin(); i != s.end(); ++i)
            cout << *i << " ";
    }
};

int main()
{
    Solution obj;
    vector<char> input{'h', 'e', 'l', 'l', 'o'};
    obj.reverseString(input);
}

File: LeetCode/String/Easy/344. Reverse String/main.py
"""
Steps to reverse strings in list-
1) Reverse last and first word
2) decrease and increase the right and left pointers respectively
"""


class Solution:
    def reverseString(self, s):
        l = 0
        r = len(s)-1
        while(l < r):
            s[r], s[l] = s[l], s[r]
            l += 1
            r -= 1
        print(s)


obj = Solution()
obj.reverseString(["h", "e", "l", "l", "o"])
